File: 905. Sort Array By Parity/Solution.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int start_index = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]%2==0){
                int temp = nums[i];
                nums[i]=nums[start_index];
                nums[start_index]=temp;
                start_index+=1;
            }
        }
        return nums;
    }
}

File: 905. Sort Array By Parity/Solution.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        lp, rp = 0, len(nums)-1
        while lp<rp:
            if nums[lp]%2 == 1:
                while nums[rp]%2!=0 and lp<rp:
                    rp-=1
                nums[rp],nums[lp] = nums[lp],nums[rp]
                rp-=1
            else:
                lp+=1
        return nums